Client/Frontend/Home/HistoryHighlights/FxHomeHistoryHighlightsVC.swift

```diff
@@ -0,0 +1,239 @@
+
+            let itemURL = item.url?.absoluteString ?? ""
+            let site = Site(url: itemURL, title: item.displayTitle, bookmarked: true)
+
+            let cellOptions = RecentlyVisitedCellOptions(title: site.title,
+                                                         shouldHideBottomLine: hideBottomLine,
+                                                         with: cornersToRound,
+                                                         and: nil,
+                                                         andIsFillerCell: false)
+
+            cell.updateCell(with: cellOptions)
+
+        } else {
+            // A filler cell
+            let cellOptions = RecentlyVisitedCellOptions(shouldHideBottomLine: hideBottomLine,
+                                                         with: cornersToRound,
+                                                         andIsFillerCell: true)
+
+            cell.updateCell(with: cellOptions)
+        }
+
+        return cell
+    }
+
+    // MARK: - Cell helper functions
+
+    /// Determines whether or not, given a certain number of items, a cell's given index
+    /// path puts it at the bottom of its section, for any matrix.
+    ///
+    /// - Parameters:
+    ///   - indexPath: The given cell's `IndexPath`
+    ///   - totalItems: The number of total items
+    /// - Returns: A boolean describing whether or the cell is a bottom cell.
+    private func isBottomCell(indexPath: IndexPath, totalItems: Int?) -> Bool {
+        guard let totalItems = totalItems else { return false }
+
+        // First check if this is the last item in the list
+        if indexPath.row == totalItems - 1
+            || isBottomOfColumn(with: indexPath.row, totalItems: totalItems)
+        { return true }
+
+        return false
+    }
+
+    private func isBottomOfColumn(with currentIndex: Int, totalItems: Int) -> Bool {
+        var bottomCellIndex: Int
+        for column in 1...numberOfColumns {
+            bottomCellIndex = (HistoryHighlightsCollectionCellConstants.maxNumberOfItemsPerColumn * column) - 1
+            if currentIndex == bottomCellIndex { return true }
+        }
+
+        return false
+    }
+
+    private func determineCornerToRound(indexPath: IndexPath, totalItems: Int?) -> UIRectCorner {
+        guard let totalItems = totalItems else { return [] }
+        var cornersToRound = UIRectCorner()
+
+        if isTopLeftCell(index: indexPath.row, totalItems: totalItems) { cornersToRound.insert(.topLeft) }
+        if isTopRightCell(index: indexPath.row, totalItems: totalItems) { cornersToRound.insert(.topRight) }
+        if isBottomLeftCell(index: indexPath.row, totalItems: totalItems) { cornersToRound.insert(.bottomLeft) }
+        if isBottomRightCell(index: indexPath.row, totalItems: totalItems) { cornersToRound.insert(.bottomRight) }
+
+        return cornersToRound
+    }
+
+    private func isTopLeftCell(index: Int, totalItems: Int) -> Bool {
```

Comment:
totalItems isn't used in this functions 🤔 I'm debating in my head if it's preferable to remove unused params, or go for the clean look of having the same params in `determineCornerToRound` lol. I feel we should remove params we don't use.